### PR TITLE
Adding support to take in a Hugging Face user token to pull models directly with authorization.

### PR DESCRIPTION
**Pull request by sourabratabose**
This pull request attempts to ad the feature as requested in isssue #6613

### CODE REVIEW

File: common/common.cpp [hunk 1]
```diff
@@ -2089,6 +2098,12 @@ struct llama_model * llama_load_model_from_url(
         return NULL;
     }
 
+    if(params.hf_token!=NULL){
```
Comment (phymbert) [inline common/common.cpp:2101]: Mind your linter. Spaces are important

File: common/common.cpp [hunk 2]
```diff
@@ -2089,6 +2098,12 @@ struct llama_model * llama_load_model_from_url(
...
+        struct curl_slist *headers = NULL;
```
Comment (phymbert) [inline common/common.cpp:2102]: Same it must be `curl_slist * headers`. Try to configure your IDE to follow exactly the rules of llama.cpp formating

File: common/common.cpp [hunk 3]
```diff
@@ -2089,6 +2098,12 @@ struct llama_model * llama_load_model_from_url(
...
+        headers = curl_slist_append(headers, "Authorization: Bearer " + params.hf_token);
```
Comment (phymbert) [inline common/common.cpp:2103]: You can do it in the line before during the initialization of the headers variable, no ?

File: common/common.cpp [hunk 4]
```diff
@@ -767,6 +767,15 @@ bool gpt_params_find_arg(int argc, char ** argv, const std::string & arg, gpt_pa
         params.hf_file = argv[i];
         return true;
     }
+    if (arg == "-hft" || arg == "--hf-token") {
+        params.hf_token = NULL;
```
Comment (phymbert) [inline common/common.cpp:771]: Where is hf_token definition?

**Review by phymbert — CHANGES_REQUESTED** [review]
Thanks for the effort. Few note:
1. Please follow the PR summary format: Motivation, Changes, Test and use  "closes #xxx". See othe PR examples.

2. Please test locally before pushing, it appears not to compile.

3. Linter are really important in c/c++, please mind every character. 

I advise you to fix on another branch, be sure all the CI jobs are working on your Fork, then updating this branch with one commit.

Good luck, please ping if not clear.

File: common/common.cpp (hunk 1 above)
Comment (TheBongStack) [reply common/common.cpp:2101]: how do I lint this code ? should I just use the linter in github workflows installed?

File: common/common.cpp (hunk 4 above)
Comment (TheBongStack) [reply common/common.cpp:771]: i cant find the struct gpt_params defination to add this.

Comment (phymbert) [reply common/common.cpp:771]: it's in common.h

File: common/common.cpp (hunk 1 above)
Comment (phymbert) [reply common/common.cpp:2101]: Which IDE are you using ?

**Pull request closed without merging**